### PR TITLE
Pass userID prop to chat adapter setUserId()

### LINKED ISSUE #2818: [Embed package] Pass userid to token generation logic
## Version
4.7.1

## Describe the bug

The logic in Embed package should pass in userid (included in query parameters) to create conversation/token generation. Today the userid and username is passed as query parameter but not included in the token. 

Could you help to enhance embed\src to include this?

[Bug]

### PR DESCRIPTION
**Pull request by timenick**
> Fixes #2818 

### Added

-  Resolves [#3544](https://github.com/microsoft/BotFramework-WebChat/issues/3544). Send user ID from props to chat adapter, by [@timenick](https://github.com/timenick) in PR [#3544)(https://github.com/microsoft/BotFramework-WebChat/issues/3544).

### Changed

-  Bumped all dependencies to the latest versions, by [@compulim](https://github.com/compulim) in PR [#3594](https://github.com/microsoft/BotFramework-WebChat/pull/3594), PR [#3694](https://github.com/microsoft/BotFramework-WebChat/pull/3694), and PR [#3544](https://github.com/microsoft/BotFramework-WebChat/pull/3544)
   -  Production dependencies
      -  [`botframework-directlinejs@0.14.1`](https://npmjs.com/package/botframework-directlinejs)

## Specific Changes

- Bumped to `botframework-directlinejs@0.14.1`
- Will now call `directLine.setUserId` if the function is available on the chat adapter
  - This will pass the user ID from Web Chat prop, to DLJS just before it connect
  - In scenarios, secret or token without user ID, developer can pass user ID thru props and the bot receiving the start conversation activity will receive the user ID

Need update the package.json and add tests after DLJS changes released.

Related to https://github.com/microsoft/BotFramework-DirectLineJS/pull/316

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [x] ~Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [x] ~Browser and platform compatibilities reviewed
-  [x] ~CSS styles reviewed (minimal rules, no `z-index`)
-  [x] ~Documents reviewed (docs, samples, live demo)
-  [x] ~Internationalization reviewed (strings, unit formatting)
-  [x] `package.json` and `package-lock.json` reviewed
-  [x] ~Security reviewed (no data URIs, check for nonce leak)
-  [x] ~Tests reviewed (coverage, legitimacy)

### CODE REVIEW

**Comment by corinagum** [comment]
Adding [DRAFT] to this since there are DLJS changes & tests blocking. @timenick please remove [DRAFT] and ping us again when you're ready.

File: packages/bundle/package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
   "dependencies": {
     "@babel/runtime": "7.12.5",
     "adaptivecards": "2.5.0",
-    "botframework-directlinejs": "0.13.1",
+    "botframework-directlinejs": "0.14.1",
```
Comment (corinagum) [inline packages/bundle/package.json:34]: Doesn't this need to be 14.2?

**Review by corinagum — APPROVED** [review]
LGTM

File: packages/bundle/package.json (hunk 1 above)
Comment (compulim) [reply packages/bundle/package.json:34]: 0.14.1 is correct, it includes both @timenick and @stevengum PR. 😉

**Comment by compulim** [comment]
Verified the behavior with @p-nagpal and we agree it looks good.

**Review by compulim — APPROVED** [review]

**Pull request merged**